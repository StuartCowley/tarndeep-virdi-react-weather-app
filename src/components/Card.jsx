import Forecast from "./Forecast";
import React from "react";
import data from "../data/forecast.json";

export default function Card() {
  let cards;

  return <section className="cards">{cards}</section>;
}
