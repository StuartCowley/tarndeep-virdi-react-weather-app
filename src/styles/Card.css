.card--forecast {
  width: 20vw;
  max-width: 300px;
  text-align: center;
}

.cards * {
  border: 1px red solid;
}

.cards {
  display: flex;
  justify-content: center;
}

@media (max-width: 550px) {
  .card--forecast {
    display: flex;
    width: 100%;
    max-width: 100%;
    justify-content: space-between;
    max-height: 80px;
  }
  .cards {
    margin: 0 auto;
    width: 90vw;
    display: block;
  }
}
