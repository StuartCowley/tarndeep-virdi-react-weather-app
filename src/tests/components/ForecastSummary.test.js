import React from "react";
import { render } from "@testing-library/react";
import Card from "../../components/ForecastSummary";

describe("Card", () => {
  const validProps = [
    {
      date: 1111111,
      description: "Stub description",
      icon: "stubIcon",
      temperature: {
        min: 12,
        max: 22,
      },
    },
    {
      date: 2222222,
      description: "Stub description2",
      icon: 602,
      temperature: {
        max: 24,
        min: 13,
      },
    },
  ];

  it("renders correctly", () => {
    const { asFragment } = render(
      <Card
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct number of instances", () => {
    const { getAllByTestId } = render(<Card userInput={validProps} />);

    expect(getAllByTestId("card--forecast")).toHaveLength(1);
  });
});
