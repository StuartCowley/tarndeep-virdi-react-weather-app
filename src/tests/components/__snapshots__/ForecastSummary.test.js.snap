// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<DocumentFragment>
  <section
    class="cards"
  >
    <p>
      Loading...
    </p>
  </section>
</DocumentFragment>
`;
